pkg/vm/vm.go

```diff
@@ -318,18 +318,22 @@ func (v *VM) execute(ctx *Context, op Instruction) {
 		v.estack.PushVal(ab)
 	case SUBSTR:
 		l := int(v.estack.Pop().BigInt().Int64())
-		o := int(v.estack.Pop().BigInt().Int64())
-		s := v.estack.Pop().Bytes()
 		if l < 0 {
 			panic("negative length")
 		}
+		o := int(v.estack.Pop().BigInt().Int64())
 		if o < 0 {
 			panic("negative index")
 		}
-		if l+o > len(s) {
-			panic("out of bounds access")
+		s := v.estack.Pop().Bytes()
+		if o > len(s) {
```

Comment:
Shouldn't it be `>=`?

Reply:
No.
https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/ExecutionEngine.cs#L457
Go behaves the same way https://play.golang.org/p/tFhXz61z_sP